mg.js

```diff
@@ -22,51 +16,147 @@ function shuffleArray(array) {
-
-      if (flippedCards.length === 2) {
+
+    if (isChecking || img.src === frontImg) return;
+
+    img.src = frontImg;
+    updateFlipCounter();
+
+    flippedCards.push({
+      button: cardButton,
+      img: img,
+    });
+
+    if (flippedCards.length === 2) {
+      isChecking = true;
+
+      const [firstCard, secondCard] = flippedCards;
+      const firstCardImgSrc = firstCard.img.src;
+      const secondCardImgSrc = secondCard.img.src;
+      if (firstCardImgSrc === secondCardImgSrc) {
+        setTimeout(() => {
+          firstCard.button.style.visibility = "hidden";
+          secondCard.button.style.visibility = "hidden";
+          flippedCards = [];
+          isChecking = false;
+
+          const allCards = document.querySelectorAll(".card");
+          const hiddenCards = Array.from(allCards).filter(
+            (card) => card.style.visibility === "hidden"
+          );
+
+          if (hiddenCards.length === allCards.length) {
+            stopTimer();
+            showResults();
+          }
+        }, 300);
+      } else {
         setTimeout(() => {
-          flippedCards.forEach((card) => {
-            const img = card.querySelector("img");
+          flippedCards.forEach(({ img }) => {
             img.src = backImg;
           });
           flippedCards = [];
+          isChecking = false;
         }, flipBackDelay);
       }
     }
-    showFront = !showFront;
   });
 
   return cardButton;
 }
-const shuffledFrontImgs = shuffleArray(doubledFrontImgs);
-const gameBoard = document.querySelector(".game-board");
-const cardContainer = document.createElement("div");
-cardContainer.className = "card-container";
-gameBoard.appendChild(cardContainer);
-shuffledFrontImgs.forEach((frontImg) => {
-  cardContainer.appendChild(createCardButton(frontImg));
+
+async function loadCards() {
+  const frontImg = await getData();
+
+  if (frontImg === null) {
+    return;
+  }
+
+  const doubledFrontImgs = [...frontImg, ...frontImg];
+  const shuffledFrontImgs = shuffleArray(doubledFrontImgs);
+  const gameBoard = document.querySelector(".game-board");
+  if (gameBoard) {
+    gameBoard.innerHTML = "";
+    const cardContainer = document.createElement("div");
+    cardContainer.className = "card-container";
+    gameBoard.appendChild(cardContainer);
+
+    shuffledFrontImgs.forEach((frontImg) => {
+      cardContainer.appendChild(createCardButton(frontImg));
+    });
+  }
+}
+
+function showResults() {
+  const elapsedTime = Math.floor((Date.now() - startTime) / 1000);
+  const resultText = `You completed the game in ${elapsedTime}s with ${flipCount} flips.`;
+  document.getElementById("result-text").textContent = resultText;
+  document.getElementById("result-modal").style.display = "block";
+}
+
+function resetGame() {
+  flipCount = 0;
+  timerStarted = false;
+  stopTimer();
+  document.getElementById("timer").textContent = "Time: 0s";
+  document.getElementById("flip-counter").textContent = "Flips: 0";
+  loadCards();
```

Comment:
```suggestion
 await loadCards();
```
`loadCards` is an async function so it should be awaited